### PR TITLE
Feature name index

### PR DESCRIPTION
**Pull request by anandthakker**
Prep for: https://github.com/geojson-xyz/geojson-xyz/issues/9
- Not actually needed, but was a side product of initial research: adds property metadata to `files.json`,  mapping field names to types
- `name-index.csv`
- Script to build index using:
  - `download-properties` script that downloads all the geojsons listed and saves only their properties
  - `write-name-index` script that uses properties downloaded from above to make a csv that relates `*name*` property values to filenames where they're found

### CODE REVIEW

**Comment by anandthakker** [comment]
@tmcw Just did some stupid rebasing and force pushing on this branch -- just a heads up, in case you'd already fetched this branch that you should fetch again.

Lemme know what you think of the initial approach here.  My thinking is that, when it wants to search by feature name, `geojson-xyz` could scan through the CSV index to find the right geojson file, and then output (a filtered version of) it.  Assuming that works, I think the lingering question is about when the index should get generated:
- Generate it & zip it up on `prepublish`, unzip on `postinstall`.  (I guess a variation of this would be to generate it and stick it on S3 alongside the geojson, and have `geojson-xyz` download it from there)
- Just generate on `postinstall` -- but this means that every user is downloading every geojson from the S3 in order to index them locally.

File: scripts/write-property-metadata [hunk 1]
```diff
@@ -0,0 +1,56 @@
+var fs = require('fs');
```
Comment (tmcw) [inline scripts/write-property-metadata:6]: this needs a hashbang

File: scripts/write-property-metadata [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  for (var cat in files) {
+    properties[cat] = {};
+    for (var sub in files[cat]) {
+      properties[cat][sub] = {};
+      files[cat][sub].forEach(function (f) {
+        try {
+          var p = JSON.parse(fs.readFileSync(path.join(propertiesDir, f)));
+          properties[cat][sub][f] = propertyTypes(p);
+        } catch (e) {
+          process.stderr.write('Error parsing properties for ' + f + '\n');
```
Comment (tmcw) [inline scripts/write-property-metadata]: should this tolerate broken files?

File: scripts/download-properties [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+var metadata = require(path.resolve(inputDir, 'meta.json'));
+var files = require(path.resolve(inputDir, 'files.json'));
+
+getProperties(function (err) { if (err) { throw err; } });
+
+// Download every geojson, but save only the properties for each feature, i.e.
+// strip geometries
+function getProperties (callback) {
+  var q = queue(5);
+  var tasks = [];
```
Comment (tmcw) [inline scripts/download-properties]: the build-list-of-tasks-and-then-defer-them technique isn't necessary, you should be able to defer in the loop itself.

**Comment by anandthakker** [comment]
@tmcw addressed your line notes (thanks for those), and updated to match 2.0 structure.

Note that `write-property-metadata` is now adding property->type map directly to files.json.  Is this worth doing?  If so, should I update `generate.js` to do it?

**Comment by anandthakker** [comment]
Err, actually after getting rid of those dumb filename duplicates, the index is only 3M, and I realized I had checked it in anyway.  Removed postinstall.

**Comment by anandthakker** [comment]
Self-merging this after slack chat w/ @tmcw

**Pull request merged**